format/src/main/scala/org/scalafmt/internal/TreePrinterUtils.scala

```diff
@@ -0,0 +1,21 @@
+package org.scalafmt.internal
+
+import scala.meta._
+
+import org.typelevel.paiges.Doc
+import org.typelevel.paiges.Doc._
+
+trait TreePrinterUtils extends WithPrinter {
+  implicit class XtensionTerms(private val terms: List[Term]) {
+    def mkDoc(start: Doc, separators: List[Doc], end: Doc): Doc = {
+      assert(terms.size == separators.size + 1)
+      val commaSeparated =
+        terms.map(print).zipAll(separators, empty, empty).foldLeft(empty) {
+          case (acc, (term, sep)) => {
+            acc + term + sep
```

Comment:
I think this would be a good candidate for a new paiges combinator.

When there is a space in the trivia, we get a double space. I would add a 
Doc.optionnalSpace. It would keep the original document if it ends with space. Otherwise, it would append the original space.

Reply:
I agree, but new combinators require design work to make sure they are as generic as possible. I'm weary of adding ad-hoc combinators to solve prettyness issues at this point. It's best to prioritize correctness for now